crates/edr_provider/src/data.rs

```diff
@@ -330,6 +337,9 @@ impl<LoggerErrorT: Debug, TimerT: Clone + TimeSinceEpoch> ProviderData<LoggerErr
             self.subscriber_callback.clone(),
             self.call_override.clone(),
             config,
+            // `hardhat_reset` doesn't discard contract metadata added with
```

Comment:
If we're doing an overhaul of `hardhat_reset` for HH3, should we reconsider the design of this too?

cc: @fvictorio